apiserver/facades/agent/credentialvalidator/register.go

```diff
@@ -24,5 +24,9 @@ func newCredentialValidatorAPI(ctx facade.ModelContext) (*CredentialValidatorAPI
 		ctx.DomainServices().Cloud(),
 		ctx.DomainServices().Credential(),
 		ctx.Resources(), ctx.Auth(),
-		ctx.Logger().Child("credentialvalidator"))
+		ctx.DomainServices().Model(),
```

Comment:
We want to pass in a WatchModelCredential closure where the model uuid is used here and not inside the facade. There are other examples of this already.

```
modelCredentialWatcher := func(stdCtx context.Context) (watcher.NotifyWatcher, error) {
    return ctx.DomainServices().Model().WatchModelCredential(stdCtx, ctx.ModelUUID())
}

```

The facade struct has an attribute
`modelCredentialWatcher func(ctx context.Context) (watcher.NotifyWatcher, error)`

Reply:
Refactored with thanks Ian! Just wondering is there a reason why we are avoiding the passing in of model uuid?